File: Python/problem1.py
def sum_multiples(m, end):
    n = (end - 1) // m
    return m * n * (n + 1) // 2


def sum_multiples_of_3_and_5(end):
    return (sum_multiples(3, end) +
            sum_multiples(5, end) -
            sum_multiples(15, end))


if __name__ == '__main__':
    print(sum_multiples_of_3_and_5(1000))

File: Haskell/Problem1.hs
module ProjectEuler.Problem1 where

sumMultiples :: Integral a => a -> a-> a
sumMultiples m end = m * n * (n + 1) `div` 2
  where
    n = (end - 1) `div` m

sumMultiplesOf3And5 :: Integral a => a -> a
sumMultiplesOf3And5 end =
  sumMultiples 3 end +
  sumMultiples 5 end -
  sumMultiples (lcm 3 5) end

main :: IO ()
main = print $ sumMultiplesOf3And5 1000
